deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py

```diff
@@ -54,7 +54,7 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         scorers = context.get_scorers(self.scorers, use_avg_defaults=False)
 
         results = []
-        classes = dataset.classes
+        classes = model.classes_ if hasattr(model, 'classes_') else dataset.classes
```

Comment:
Is this a hot fix? if not then it should be done in the context level not inside a specific check